Sources/TuistSigning/SigningCipher.swift

```diff
@@ -0,0 +1,134 @@
+                try decryptData($0, masterKey: masterKey)
+            }
+
+        try zip(decipheredKeys, signingKeyFiles).forEach {
+            try $0.write(to: $1.url)
+        }
+    }
+
+    // MARK: - Helpers
+
+    /// Encrypts `data`
+    /// - Parameters:
+    ///     - data: Data to encrypt
+    ///     - masterKey: Master key data
+    /// - Returns: Encrypted data
+    private func encryptData(_ data: Data, masterKey: Data) throws -> Data {
+        let iv = try generateIv()
+        let aesCipher = try AES(key: masterKey.bytes, blockMode: CTR(iv: iv.bytes), padding: .noPadding)
+        guard
+            let encryptedBase64String = try aesCipher.encrypt(data.bytes).toBase64(),
+            let data = (iv.base64EncodedString() + "-" + encryptedBase64String).data(using: .utf8)
+        else { throw SigningCipherError.failedToEncrypt }
+        return data
+    }
+
+    /// Decrypts `data`
+    /// - Parameters:
+    ///     - data: Data to decrypt
+    ///     - masterKey: Master key data
+    /// - Returns: Decrypted data
+    private func decryptData(_ data: Data, masterKey: Data) throws -> Data {
+        guard
+            let encodedString = String(data: data, encoding: .utf8),
+            let dividerIndex = encodedString.firstIndex(of: "-"),
+            let iv = Data(base64Encoded: String(encodedString.prefix(upTo: dividerIndex)))
+        else { throw SigningCipherError.failedToEncrypt }
+
+        let dataToDecrypt = Data(base64Encoded: String(encodedString.suffix(from: dividerIndex).dropFirst()))
+        let aesCipher = try AES(key: masterKey.bytes, blockMode: CTR(iv: iv.bytes), padding: .noPadding)
+        guard let decryptedData = try dataToDecrypt?.decrypt(cipher: aesCipher) else { throw SigningCipherError.failedToEncrypt }
+        return decryptedData
+    }
+
+    /// - Returns: Files we want encrypt/decrypt along with master key data
+    private func signingData(at path: AbsolutePath) throws -> (signingKeyFiles: [AbsolutePath], masterKey: Data) {
+        guard
+            let rootDirectory = rootDirectoryLocator.locate(from: path)
+        else { throw SigningCipherError.rootDirectoryNotFound(path) }
+        let signingDirectory = rootDirectory.appending(components: Constants.tuistDirectoryName, Constants.signingDirectoryName)
+        let masterKey = try self.masterKey(from: signingDirectory)
+        // Find all files in `signingDirectory` with the exception of "master.key"
+        let signingKeyFiles = FileHandler.shared.glob(signingDirectory, glob: "**/*")
+            .filter { $0.pathString != signingDirectory.appending(component: "master.key").pathString }
+            .filter { !FileHandler.shared.isFolder($0) }
+        return (signingKeyFiles: signingKeyFiles, masterKey: masterKey)
+    }
+
+    /// - Returns: Master key data
+    private func masterKey(from signingDirectory: AbsolutePath) throws -> Data {
+        let masterKeyFile = signingDirectory.appending(component: "master.key")
```

Comment:
is it worth creating a constant for this, since it's used in two places and is critical to the algorithm

Reply:
Done 👍